### PR TITLE
[glsl-out] Convert modulo operator on float to SPIR-V `OpFRem` equivalent function

### LINKED ISSUE #1451: [wgsl-in]/[glsl-out] Bug with vec types and modulo operator
I had some code in wgsl like this:
```rust
    let rgb = clamp(
        abs(
            ((hue * 6.0 + vec3<f32>(0.0, 4.0, 2.0)) % 6.0 - 3.0
        ) - 1.0,
        vec3<f32>(0.0),
        vec3<f32>(1.0)
    );
```
and hit this when trying to run on WebGL2:
```
Internal error in VERTEX shader: ERROR: 0:244: '%' : wrong operand types - no operation '%' exists that takes a left-hand operand of type 'highp 3-component vector of float' and a right operand of type 'const 3-component vector of float' (or there is no acceptable conversion)
```
It's specifically the calculated vec3 % constant vec3 that seems to be the problem. I can work around it by individually calculating each component, but I think this is a bug somewhere, perhaps in glsl-out?

### PR DESCRIPTION
**Pull request by Gordon-F**
Fix #1451

### CODE REVIEW

**Comment by JCapucho** [comment]
This translation isn't correct see #1410

File: src/back/glsl/mod.rs [hunk 1]
```diff
@@ -311,6 +311,16 @@ pub enum Error {
     Custom(String),
 }
 
+/// Binary operation with a different logic on the GLSL side
+enum BinaryOperation {
```
Comment (kvark) [inline src/back/glsl/mod.rs:315]: that's nice

File: src/back/glsl/mod.rs [hunk 2]
```diff
@@ -2214,35 +2224,79 @@ impl<'a, W: Write> Writer<'a, W> {
...
-                write!(self.out, "{}(", function.unwrap_or(""))?;
-                self.write_expr(left, ctx)?;
-
-                if function.is_some() {
-                    write!(self.out, ",")?
-                } else {
-                    write!(self.out, " {} ", super::binary_operation_str(op))?;
+                match function {
+                    BinaryOperation::VectorCompare => {
+                        match op {
```
Comment (kvark) [inline src/back/glsl/mod.rs:2261]: let's move the `write!` out of the match, which can return just an `&str`

File: src/back/glsl/mod.rs [hunk 3]
```diff
@@ -2214,35 +2224,79 @@ impl<'a, W: Write> Writer<'a, W> {
...
+                            Bo::NotEqual => write!(self.out, "notEqual")?,
+                            _ => {}
+                        };
+                    }
+                    BinaryOperation::Modulo | BinaryOperation::Other => {
+                        // nothing to do here
+                    }
+                }
+                write!(self.out, "(")?;
+                match function {
```
Comment (kvark) [inline src/back/glsl/mod.rs:2276]: let's merge this `match` with the previous one

**Review by kvark — APPROVED** [review]

**Pull request merged**